File: player/ultraPlayer/WaveformToStringConverter.cpp
#include "WaveformToStringConverter.h"

WaveformToStringConverter::WaveformToStringConverter(void)
{
}

WaveformToStringConverter::~WaveformToStringConverter(void)
{
}

std::string WaveformToStringConverter::Convert( WaveformType::Type data )
{
	throw NotImplementedException("", "WaveformToStringConverter::Convert");
}

WaveformType::Type WaveformToStringConverter::Convert( string data )
{
	WaveformType::Type type;

	// convert it to number
	int degrees = NumericToStringConverter::Convert<int>(data);

	try
	{
		if (data == CommunicationProtocol::Instance()->SineWaveformParameter())
		{
			type = WaveformType::Sine;
		}
		else if (data == CommunicationProtocol::Instance()->SquareWaveformParameter())
		{
			type = WaveformType::Square;
		}
		else if (data == CommunicationProtocol::Instance()->TriangleWaveformParameter())
		{
			type = WaveformType::Triangle;
		}
		else
		{
			throw ConverterException("unsupported waveform","WaveformToStringConverter::Convert");
		}
	}
	catch (exception& e)
	{
		throw;
	}

	return type;
}

File: controller/Model/SoundFx/WaveformToStringConverter.cs
using System;
using System.Collections.Generic;
using System.Text;
using UltraPlayerController.Resources.Communication;
using Utilities.Convert;

namespace UltraPlayerController.Model.SoundFx
{
    public class WaveformToStringConverter: IConverter<string, WaveformType>
    {
        #region IConverter<string,WaveformType> Members

        public string Convert(WaveformType data)
        {
            string result = string.Empty;

            try
            {
                switch (data)
                {
                    case WaveformType.Sine:
                        result = Protocol.Instance.SineWaveformParameter;
                        break;
                    case WaveformType.Square:
                        result = Protocol.Instance.SquareWaveformParameter;
                        break;
                    case WaveformType.Triangle:
                        result = Protocol.Instance.TriangleWaveformParameter;
                        break;
                    default:
                        break;
                }
            }
            catch (System.Exception)
            {
            	
            }

            return result;
        }

        public WaveformType Convert(string data)
        {
            throw new NotImplementedException();
        }

        #endregion
    }
}
